breakingchange: override sep.characters

CarstenKoenig opened the pull request:
This is a breaking change as it changes the `Data.Formatter.Number.Formatter` record.

**Description of the change**

It adds support to have different characters for decimal- and thousand-group-separators.
A use case would be formatting numbers in Germany as there these characters are flipped (1.234,56 instead of 1,234.56).

It also fixes a bug in the parsing of numbers with `comma=true` in the `Formatter` (`1.12` was rejected because `1` was only a single digit not 3 - this check now only happens after the first thousand-group-separator character was parsed.

---

**Checklist:**

- [x] Added the change to the changelog's "Unreleased" section with a link to this PR and your username
- [x] Linked any existing issues or proposals that this pull request should close
- [x] Updated or added relevant documentation in the README and/or documentation directory
- [x] Added a test for the contribution (if applicable)

CarstenKoenig commented:
This should address #71

CarstenKoenig commented:
Hi - I hope I checked all the boxes correctly ;)

garyb commented on src/Data/Formatter/Number.purs:
```diff
@@ -41,8 +43,18 @@ newtype Formatter = Formatter
+-- |
+-- | > parseFormatString ".." # map (changeSeparators { groupSeparator: '.', decimalSeparator: ','})
+changeSeparators :: { groupSeparator :: Char, decimalSeparator :: Char } -> Formatter -> Formatter
+changeSeparators { groupSeparator, decimalSeparator } (Formatter formatter) =
```
I'd suggest calling this `withSeparators` perhaps? But other than that this looks good to me!

CarstenKoenig replied:
no problem - guess it sounds better on use-side this way - I'll change it

CarstenKoenig commented:
Hi,

I think the CI did break because the package.dhall seems to misformfed/invalid - I noticed this for the PR but did not want to change the file (as I did not know if you had some other pipeline in place that needs this format)

Locally I renamed the spago.dhall, deleted the package.dhall redid `spago init` and then copied the old spago.dhall over - this did the trick locally (tests did pass).

If you like I'll happily add both files to the PR

garyb commented:
No worries - I was about to fix this up and push into the branch myself, it just needs the package-set location updating to one that actually exists (seems we deleted the prepare-0.15 after 0.15 was released). Sorry about the slow back and forth on this by the way!

CarstenKoenig commented:
I'm glad I can help out - take your time.

CarstenKoenig commented:
Thank you :smile:

garyb commented:
I'm hoping to get #74 included in the breaking release, so it might be another day or two before that goes in, but will definitely get it out this week. Thanks very much for the contribution!
